core/src/main/java/hudson/PluginManager.java

```diff
@@ -1313,20 +1311,18 @@ public HttpResponse doInstallPlugins(StaplerRequest req) throws IOException {
             installJobs.add(jobFuture);
         }
 
-        if (Jenkins.getActiveInstance().getInstallState() == InstallState.NEW) {
-            trackInitialPluginInstall(installJobs);
-        }
+        trackInitialPluginInstall(installJobs);
 
         return installJobs;
     }
 
     private void trackInitialPluginInstall(@Nonnull final List<Future<UpdateCenter.UpdateCenterJob>> installJobs) {
-        final Jenkins jenkins = Jenkins.getActiveInstance();
+        final Jenkins jenkins = Jenkins.getInstance();
         final UpdateCenter updateCenter = jenkins.getUpdateCenter();
 
-        updateCenter.persistInstallStatus();
-        if (jenkins.getInstallState() == InstallState.NEW) {
+        if (!Jenkins.getInstance().getInstallState().isSetupComplete()) {
             jenkins.setInstallState(InstallState.INITIAL_PLUGINS_INSTALLING);
```

Comment:
OK, so why do we need separate NEW and INITIAL_PLUGINS_INSTALLING states if we are  moving between them outside of InstallState flow.